DataScience/AzureStorageDownloader.py

```diff
@@ -1,127 +1,176 @@
                  "--------------------------------------------------------------------\n"
                  "Notes:\n"
-                 "end_date is not included\n"
+                 "data from end_date is not included\n"
                  "overwrite: would replace files on disk\n"
                  "dry_run: print which blobs would have been downloaded, without downloading\n"
-                 "skip_current: if true, avoid downloading a blob which has been modified within the last 30 secs\n"
-                 "--------------------------------------------------------------------")
-    if len(sys.argv) < 2:
-        print(usage_str)
-        print("Example Usage: python AzureStorageDownloader.py cmplx --start_date 2017-10-15 --end_date 2017-10-18")
-        sys.exit()
-        
-    config = configparser.ConfigParser()
-    config.read('ds.config')
-    visua_config = config['Visualization']
-    
-    log_dir = visua_config['LogDir']
-    pkl_data_fp = visua_config['PickleDataPath']
-    
-            
-    container = sys.argv[1]
-    print('Container: {0}'.format(container))
+                 "skip_current: if true, avoid downloading a blob which has been modified within the last 10 min\n"
+                 "version: integer describing which version of data downloader to use (default: 2 -> AzureStorageDownloader)"
+                 "auth_fp: file path of pickle file containing azure storage authentication (when this flag is missing, ds.config info is used for authentication\n"
+                 "--------------------------------------------------------------------".format(argv[0]))
 
-    ################################# LOG DOWNLOADER #########################################################
-    
-    overwrite = False    
-    dry_run = False
-    start_date = None
-    end_date = None
-    skip_current = False
+    kwargs = {}
     try:
-        if len(sys.argv) > 2:
-            ii = 2
-            while ii < len(sys.argv):
-                if sys.argv[ii] == '--overwrite':
-                    overwrite = True
-                elif sys.argv[ii] == '--dry_run':
-                    dry_run = True
-                elif sys.argv[ii] == '--skip_current':
-                    skip_current = True
-                elif sys.argv[ii] == '--start_date':
-                    ii += 1
-                    start_date = datetime.datetime.strptime(sys.argv[ii], '%Y-%m-%d')
-                elif sys.argv[ii] == '--end_date':
-                    ii += 1
-                    end_date = datetime.datetime.strptime(sys.argv[ii], '%Y-%m-%d')
-                else:
-                    print('Input arg: {} not recognized! Abort!'.format(sys.argv[ii]))
-                    print(usage_str)
-                    sys.exit()
+        ii = 1
+        while ii < len(argv):
```

Comment:
any reason not to use lib/argparse or lib/optparse?

Reply:
no reason. I can do argparse